internal/zinc-persist/src/main/java/sbt/inc/ReadMapper.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Zinc - The incremental compiler for Scala.
+ * Copyright 2011 - 2017, Lightbend, Inc.
+ * Copyright 2008 - 2010, Mark Harrah
+ * This software is released under the terms written in LICENSE.
+ */
+
+package sbt.inc;
+
+import sbt.internal.inc.mappers.RelativeReadMapper;
+import xsbti.compile.MiniSetup;
+import xsbti.compile.analysis.Stamp;
+
+import java.io.File;
+import java.nio.file.Path;
+
+/**
+ * Defines a reader-only mapper interface that is used by Zinc before read
+ * the contents of the analysis files from the persistent storage.
```

Comment:
It is really not used "when" reading analysis, but "after" reading the contents.